include/lsst/afw/geom/SpanSet.h

```diff
@@ -0,0 +1,252 @@
+// -*- lsst-c++ -*-
+
+/*
+ * LSST Data Management System
+ * Copyright 2008-2016  AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_GEOM_SPANSET_H
+#define LSST_AFW_GEOM_SPANSET_H
+
+#include <vector>
+#include <algorithm>
+#include <functional>
+#include "lsst/afw/geom/Span.h"
+#include "lsst/afw/geom/Box.h"
+#include "lsst/afw/geom/SpanSetFunctorGetters.h"
+
+namespace lsst { namespace afw { namespace geom {
+
+enum class Stencil { CIRCLE, BOX, MANHATTAN };
+
+class SpanSet {
+ public:
+    typedef std::vector<Span>::const_iterator const_iterator;
```

Comment:
We should use the STL lowercase form here.  I believe this exception the our guidelines is broadly recognized, but it may need to be made explicit if it isn't already.